### PR TITLE
Make final docker image lighter

### LINKED ISSUE #17930: Docker Filesize v9.23.4
### Describe the Bug

Hi - seems that Docker Image size went up heavily now from  ~ 400MB  to  ~650MB.

Suggest to check if anything happend there...

### To Reproduce

Have a look at Docker image size

### Hosting Strategy

Self-Hosted (Docker Image)

### PR DESCRIPTION
**Pull request by rijkvanzanten**
Closes #17930

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -47,15 +48,7 @@ ENV STORAGE_LOCAL_ROOT="/directus/uploads"
...
-COPY --from=pruned /workspace/pruned/cli.js cli.js
-COPY --from=pruned /workspace/pruned/dist dist
-COPY --from=pruned /workspace/pruned/package.json package.json
-COPY --from=pruned /workspace/pruned/node_modules node_modules
-
-RUN mkdir /directus/database /directus/extensions /directus/uploads \
-	&& chown -R node:node /directus \
-	&& corepack enable \
-	&& corepack prepare pnpm@7.30.0 --activate
+COPY --from=pruned --chown=node:node /workspace/pruned .
```
Comment (rijkvanzanten) [inline Dockerfile:51]: The `pruned` folder already contains just the build essentials, so we don't have to manually chose individual files

**Comment by rijkvanzanten** [comment]
The one thing we'll have to confirm post-merge is whether or not this issue is a local quirk, or something I caused in this change:

```
#17 56.79 app build: ✓ 4876 modules transformed.
#17 59.06 app build: rendering chunks...
#17 62.14 app build: Failed
#17 62.15 undefined
#17 62.15 /workspace/app:
#17 62.15  ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  @directus/app@9.23.4 build: `vite build`
#17 62.15 Exit status 129
```

@paescuj mind taking a look on your end? Ty! 🙏🏻

File: Dockerfile [hunk 2]
```diff
@@ -21,15 +21,16 @@ RUN pnpm install --recursive --offline --frozen-lockfile
...
 WORKDIR /workspace
-ENV NODE_OPTIONS=--max-old-space-size=8192
+ENV NODE_OPTIONS=--max-old-space-size=6144
 
-RUN pnpm --recursive run build \
-	&& pnpm --filter directus deploy --prod pruned \
-	&& cd pruned \
+RUN pnpm --recursive --workspace-concurrency=1 run build
+RUN pnpm --filter directus deploy --prod pruned
+RUN cd pruned \
```
[the remark this answers is not in the document]
Comment (paescuj) [reply Dockerfile:24]: Good point 🚀

File: Dockerfile (hunk 1 above)
Comment (paescuj) [reply Dockerfile:47]: Love it ❤️🏅

**Review by paescuj — APPROVED** [review]
Looks great 👍

**Comment by rijkvanzanten** [comment]
My mac is consistently failing every build no matter the checked out commit, so I think it's something on my end locally 🤷🏻

**Pull request merged**